Add support for brush entities

EIREXE opened the pull request:
It will automatically decide if it will build collision or not based on the target entity's class being a child of PhysicsBody3D or not.

codecat commented:
Thanks for the PR! There are a lot of unnecessary style changes though (for example, `mesh_builder.cpp` is entirely re-formatted). This makes looking at the diff for this PR kinda difficult.

I also noticed that this adds functionality that is kinda similar to #10, and that the code for this system was getting a little bit out of hand. So, I've cleaned up the code by removing the (old, broken) CSGBuilder, and added the functionality of this PR. (bc080f0131be14dffb36e631a36541b283dfb756) Please let me know if this is sufficient for you. Otherwise we can always open another issue or PR. 😊

Maybe @Macronic or @kherchel can confirm whether my recent commits still work for their custom Area3D entities? 👀

EIREXE commented:
> Thanks for the PR! There are a lot of unnecessary style changes though (for example, `mesh_builder.cpp` is entirely re-formatted). This makes looking at the diff for this PR kinda difficult.
> 
> I also noticed that this adds functionality that is kinda similar to #10, and that the code for this system was getting a little bit out of hand. So, I've cleaned up the code by removing the (old, broken) CSGBuilder, and added the functionality of this PR. ([bc080f0](https://github.com/codecat/godot-tbloader/commit/bc080f0131be14dffb36e631a36541b283dfb756)) Please let me know if this is sufficient for you. Otherwise we can always open another issue or PR. blush
> 
> Maybe @Macronic or @kherchel can confirm whether my recent commits still work for their custom Area3D entities? eyes

sure, looks ok, I wonder why the formatting ended up like that
